backend/moonstreamapi/providers/moonworm_provider.py

```diff
@@ -0,0 +1,453 @@
+        #         if provider_type != event_type:
+        #             continue
+
+        #         if raw_filter.startswith("from:"):
+        #             address = raw_filter[from_prefix_length:]
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.from_addresses.append(address)
+        #         elif raw_filter.startswith("to:"):
+        #             address = raw_filter[to_prefix_length:]
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.to_addresses.append(address)
+        #         else:
+        #             address = raw_filter
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.from_addresses.append(address)
+        #                 parsed_filters.to_addresses.append(address)
+
+        if not (parsed_filters.addresses):
+            return None
+
+        return parsed_filters
+
+    def stream_boundary_validator(
+        self, stream_boundary: data.StreamBoundary
+    ) -> data.StreamBoundary:
+        """
+        Stream boundary validator for the events provider.
+
+        Checks that stream boundaries do not exceed periods of greater than 24 hours.
+
+        Raises an error for invalid stream boundaries, else returns None.
+        """
+        valid_period_seconds = 24 * 60 * 60
+
+        _, stream_boundary = validate_stream_boundary(
+            stream_boundary, valid_period_seconds, raise_when_invalid=True
+        )
+        return stream_boundary
+
+    def query_events(
+        self,
+        db_session: Session,
+        stream_boundary: data.StreamBoundary,
+        parsed_filters: Filters,
+    ) -> Query:
+        """
+        Builds a database query for Ethereum transactions that occurred within the window of time that
+        the given stream_boundary represents and satisfying the constraints of parsed_filters.
+        """
+
+        Labels = get_label_model(self.blockchain)
+
+        query = db_session.query(
+            Labels.block_number,
+            Labels.address,
+            Labels.transaction_hash,
+            Labels.label_data,
+            Labels.block_timestamp,
+            Labels.log_index,
+            Labels.created_at,
+        ).filter(Labels.label == "moonworm")
+
+        if stream_boundary.include_start:
+            query = query.filter(Labels.block_timestamp >= stream_boundary.start_time)
+        else:
+            query = query.filter(Labels.block_timestamp > stream_boundary.start_time)
+
+        if stream_boundary.end_time is not None:
+            if stream_boundary.include_end:
+                query = query.filter(Labels.block_timestamp <= stream_boundary.end_time)
+            else:
+                query = query.filter(Labels.block_timestamp <= stream_boundary.end_time)
+
+        addresses_filters = []
+
+        for address_filter in parsed_filters.addresses:
```

Comment:
I think the semantics are wrong here - it is possible for a user to want to query by labels without specifying an address. For example, if I wanted to query `Transfer` events for *all* ERC721 contracts in my subscriptions.

Reply:
@zomglings  "Transfer events for all ERC721 contracts in my subscriptions." Have we label ERC721 on contracts for polygon table?

Reply:
Ahh okay understand i not sure if it's must be part of moonworm provider.